source/val/validate.cpp

```diff
@@ -166,14 +137,11 @@ spv_result_t ProcessInstruction(void* user_data,
     _.AddFunctionCallTarget(inst->words[3]);
   }
 
-  DebugInstructionPass(_, inst);
-  if (auto error = CapabilityPass(_, inst)) return error;
-  // The IdPass check registers instructions and, therefore, must be called
-  // before any instruction lookups are performed.
-  if (auto error = IdPass(_, inst)) return error;
-
-  const Instruction* instruction = &(_.ordered_instructions().back());
+  auto* instruction = _.AddOrderedInstruction(inst);
```

Comment:
Or we could put this at line 122 beside the _.increment_instruction_count()....  but that also looks like the counter could be replaced with a size() call on the ordered instructions list.    (Likely the counter came first, and saving the instructions came much later.)

No change required for this review.